### PR TITLE
port changes from lightning

### PR DESCRIPTION
**Pull request by timmysilv**
Identical to PennyLaneAI/pennylane-lightning-kokkos#54

### CODE REVIEW

File: tests/test_adjoint_jacobian.py [hunk 1]
```diff
@@ -1079,3 +1105,41 @@ def circuit(params):
...
+        ),
+        (
+            qml.SparseHamiltonian(qml.Hamiltonian([1], [qml.PauliZ(0)]).sparse_matrix(), wires=[0]),
+            SparseHamiltonianGPU_C64,
+        ),
+    ],
+)
+def test_obs_returns_expected_type(obs, obs_type):
+    """Tests that observables get serialized to the expected type."""
+    assert isinstance(plk._serialize._serialize_ob(obs, dict(enumerate(obs.wires)), True), obs_type)
```
Comment (AmintorDusko) [inline tests/test_adjoint_jacobian.py:1129]: Hi, @timmysilv. This `plk` is coming from the suite of tests for PennyLane Lightning Kokkos.
This (and all others) will need to be updated for this repository.

Comment (AmintorDusko) [reply tests/test_adjoint_jacobian.py]: Looks like @multiphaseCFD is implementing the necessary changes. Forget about this comment.

Comment (timmysilv) [reply tests/test_adjoint_jacobian.py]: oops sorry about that! I swear I had fixed the copy-paste locally but it's been lost. ty for the fix!

**Review by mlxd — APPROVED** [review]
Thanks @timmysilv 
I made some subtle changes to better support the multi-GPU decomposition of H terms, and changed the tests to avoid the quirks of executions not involving the QNode. Happy for this to go in now.

**Review by AmintorDusko — APPROVED** [review]
Nice work on that @timmysilv and @mlxd. Happy to approve!

File: tests/test_adjoint_jacobian.py [hunk 2]
```diff
@@ -377,6 +388,38 @@ def test_provide_starting_state(self, tol, dev_gpu):
...
+    @pytest.mark.parametrize(
+        "old_obs",
+        [
+            qml.PauliX(0) @ qml.PauliZ(1),
+            qml.Hamiltonian([1.1], [qml.PauliZ(0)]),
+            qml.Hamiltonian([1.1, 2.2], [qml.PauliZ(0), qml.PauliZ(1)]),
+            qml.Hamiltonian([1.1, 2.2], [qml.PauliX(0), qml.PauliZ(0) @ qml.PauliX(1)]),
+        ],
+    )
+    def test_op_arithmetic_is_supported(self, old_obs, dev_gpu, tol):
```
Comment (timmysilv) [inline tests/test_adjoint_jacobian.py:400]: the parametrized `old_obs` is unused now, right?
```suggestion
    def test_op_arithmetic_is_supported(self, dev_gpu, tol):
```

Comment (mlxd) [reply tests/test_adjoint_jacobian.py:400]: Yep, good catch. The overwriting definition can be removed. I'll rerun on a GPU node and push it up

Comment (mlxd) [reply tests/test_adjoint_jacobian.py:400]: Done. Once the CI passes here (we may need to rerun them manually) we can merge.

Comment (AmintorDusko) [reply tests/test_adjoint_jacobian.py:400]: Nice!

**Pull request merged**